Automattic/jetpack
# Boost: Refactor AdvancedCriticalCss and several other components to React

## Proposed changes:

* Refactor AdvancedCriticalCss and its related components to react.

### Other information:


## Jetpack product discussion



pc9hqz-28M-p2
pc9hqz-2jy-p2

## Does this pull request change what data or activity we track or use?



no

## Testing instructions:






* Test the Advanced Recommendations page to make sure it behaves correctly.

I find it hard to cause the section to show up, so I used this mu-plugin. Just start generating the critical css with more than 4 pages and you should be good to go (or adjust the number):

```php
<?php

add_filter( 'rest_pre_dispatch', function ( $result, $server, $request ) {
	if ( '/jetpack-boost/v1/critical-css/start' === $request->get_route() ) {
		update_option( 'test_errors', 0 );
	}

	return $result;
}, 10, 3 );

add_action( 'template_redirect', function () {
	$test_errors = get_option( 'test_errors', 0 );
	$test_errors++;
	update_option( 'test_errors', $test_errors );

	if ( $test_errors >= 4 ) {
		nonexistentfunction();
	}
} );
```
